TrackletAlgorithm/ProjectionCalculator.h

```diff
@@ -237,11 +250,15 @@ template<
     nproj_disk[i] = 0;
   }
 
-  const int TCID = ((TrackletProjection<BARRELPS>::TProjTCID(Seed) << 4) + iTC);
-  const uint32_t MaskBarrel = TPROJMaskBarrel<Seed, iTC>();
-  const uint32_t MaskDisk = TPROJMaskDisk<Seed, iTC>();
+  //iTC gives you the first TCID in e.g. ABC, ipage gives you corrected TCID for B and C
+  const int TCID = ((TrackletProjection<BARRELPS>::TProjTCID(Seed) << 4) + iTC + ipage);
   bool addL3 = false, addL4 = false, addL5 = false, addL6 = false; // whether a projection has been added successfully
 
+  //FIXME
+  //ap_uint<7> trackletId = trackletIndex&31;
+  ap_uint<9> trackletId = trackletIndex;
+  trackletIndex =  trackletIndex&127;
```

Comment:
@aryd Please remove the magic numbers here, e.g., if `trackletIndex` is defined as an `ap_uint<kNBits_MemPage+kNBits_MemAddr>` (i.e., `ap_uint<9>`) instead, you could do the following:
```suggestion
  const auto trackletId = trackletIndex;
  trackletIndex =  trackletIndex & ((1 << kNBits_MemAddr) - 1);
```

Reply:
Fixed.